src/passes/Outlining.cpp

```diff
@@ -13,23 +13,225 @@
 #define DBG(statement)
 #endif
 
+// Check a Result or MaybeResult for error and call Fatal() if the error exists.
+#define ASSERT_ERR(val)                                                        \
+  if (auto _val = (val); auto err = _val.getErr()) {                           \
+    Fatal() << err->msg;                                                       \
+  }
+
 namespace wasm {
 
+struct ReconstructStringifyWalker
+  : public StringifyWalker<ReconstructStringifyWalker> {
+
+  ReconstructStringifyWalker(Module* wasm)
+    : existingBuilder(*wasm), outlinedBuilder(*wasm) {
+    this->setModule(wasm);
+    DBG(std::cout << "\n\nexistingBuilder: " << &existingBuilder
+                  << " outlinedBuilder: " << &outlinedBuilder);
+  }
+
+  // As we reconstruct the IR during outlining, we need to know whether the
+  // expression being visited during walking is:
+  //
+  // - NotInSeq (meaning not part of a
```

Comment:
Formatting looks sus on these. It also would be good to move the comments down into the enum declaration rather than repeating `NotInSeq`, etc. in the comment as well as in the code.

Reply:
done